### PR TITLE
docs(vvars): fix wrong lua types in vim.v variables

### LINKED ISSUE #26912: vim.v.stderr is `number`, not `string`
### Problem

SSIA
https://github.com/neovim/neovim/blob/28c183b38cdc00436d95af80ee8e34e4793ee38d/runtime/lua/vim/_meta/vvars.lua#L627-L628

### Steps to reproduce

any

### Expected behavior

any

### Neovim version (nvim -v)

NVIM v0.10.0-dev-28c183b

### Vim (not Nvim) behaves the same?

no

### Operating system/version

any

### Terminal name/version

any

### $TERM environment variable

any

### Installation

any

### PR DESCRIPTION
**Pull request by wookayin**
Fixup to #26682
Fixes #26912 

- cmdarg: string[] -> string
- shell_error: string -> int
- stderr: string -> int

Also add types for: cmdbang, errors, exception, false, true

### CODE REVIEW

File: src/nvim/vvars.lua [hunk 1]
```diff
@@ -229,6 +232,7 @@ M.vars = {
     ]=],
   },
   ['false'] = {
+    type = 'boolean',
```
Comment (lewis6991) [inline src/nvim/vvars.lua:235]: This isn't actually a Boolean value.

Comment (wookayin) [reply src/nvim/vvars.lua:235]: `type(vim.v['false'])` gives boolean and `vim.v['false'] == false`. Could you please elaborate?

**Comment by wookayin** [comment]
Another (minor) side question: should I title this as `docs(vvars)` rather than `docs(eval)`, as vvars is separated out of eval in #26882?

**Comment by dundargoc** [comment]
> Another (minor) side question: should I title this as `docs(vvars)` rather than `docs(eval)`, as vvars is separated out of eval in #26882?

Doesn't matter, go with what you feel is best.

**Comment by clason** [comment]
whatever sparks joy :)

File: src/nvim/vvars.lua (hunk 1 above)
Comment (lewis6991) [reply src/nvim/vvars.lua:235]: Hmm I must be mistaken.

**Pull request merged**